src/connector/src/sink/iceberg/jni_catalog.rs

```diff
@@ -0,0 +1,189 @@
+            })?;
+
+            let result_json =
+                call_method!(env, self.java_catalog.as_obj(), {String updateTable(String)},
+                &request_jni_str)
+                .with_context(|| {
+                    format!(
+                        "Failed to update iceberg table: {}",
+                        update_table.table_name()
+                    )
+                })?;
+
+            let rust_json_str = jobj_to_str(env, result_json)?;
+
+            let response: CommitTableResponse = serde_json::from_str(&rust_json_str)?;
+
+            tracing::info!(
+                "Table metadata location of {} is {}",
+                update_table.table_name(),
+                response.metadata_location
+            );
+
+            let table_metadata = response.metadata()?;
+
+            let args = IcebergTableIoArgs::builder_from_path(&table_metadata.location)?
+                .with_args(self.config.table_io_configs.iter())
+                .build()?;
+            let table_op = args.create()?;
+
+            Ok(Table::builder_from_catalog(
+                table_op,
+                self.clone(),
+                table_metadata,
+                update_table.table_name().clone(),
+            )
+            .build()?)
+        })
+        .map_err(|e| {
+            icelake::Error::new(ErrorKind::Unexpected, "Failed to update iceberg table.")
+                .set_source(e)
+        })
+    }
+}
+
+impl JniCatalog {
+    pub fn build(
+        base_config: BaseCatalogConfig,
+        name: impl ToString,
+        catalog_impl: impl ToString,
+        java_catalog_props: HashMap<String, String>,
+    ) -> Result<CatalogRef> {
+        let jvm = JVM.get_or_init()?;
+
+        execute_with_jni_env(jvm, |env| {
+            // Convert props to string array
+            let props = env.new_object_array(
+                (java_catalog_props.len() * 2) as i32,
+                "java/lang/String",
+                JObject::null(),
+            )?;
+            for (i, (key, value)) in java_catalog_props.iter().enumerate() {
+                let key_j_str = env.new_string(key).unwrap();
+                let value_j_str = env.new_string(value).unwrap();
+                env.set_object_array_element(&props, i as i32 * 2, key_j_str)?;
+                env.set_object_array_element(&props, i as i32 * 2 + 1, value_j_str)?;
+            }
+
+            let jni_catalog_wrapper = env
+                .call_static_method(
+                    "com/risingwave/connector/catalog/JniCatalogWrapper",
+                    "create",
+                    "(Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;)V",
+                    &[
+                        (&env.new_string(name.to_string()).unwrap()).into(),
+                        (&env.new_string(catalog_impl.to_string()).unwrap()).into(),
+                        (&props).into(),
+                    ],
+                )
+                .unwrap();
```

Comment:
use `call_static_method!` marco in https://github.com/risingwavelabs/risingwave/blob/3081541a954d861da6ab73139d4d7a3b26adf2fe/src/jni_core/src/macros.rs  instead?